include/deal.II/grid/grid_in.h

```diff
@@ -521,7 +521,7 @@ class GridIn
   void
   read_msh(std::istream &in);
 
-#ifdef DEAL_II_GMSH_WITH_API
+#if defined(DEAL_II_WITH_GMSH) && defined(DEAL_II_GMSH_WITH_API)
```

Comment:
I don't understand. I would expect the second define only be true if the first one is. Why can we not fix this during configure time or inside config.h?

Reply:
This is now fixed during configure time. I just added a second commit that guards the ifdefs with `DEAL_II_WITH_GMSH` as well.

Reply:
So you are saying that this change is not necessary to work? Then I don't see the point of making it here. Do we do this incorrectly for other packages?

Reply:
@tjhei At every other place we guard with `DEAL_II_WITH_*` but not for GMSH. But *meh* I'll drop the commit.